### PR TITLE
ODS-5551 Move EdFi.Ods.CodeGen Package build to Github Actions

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

**Review by bptillman — APPROVED** [review]
This looks good so far. Just a few small tweaks to make and then this should be good to merge.

File: .github/workflows/Packages- EdFi.Ods.CodeGen.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
+name: Packages- EdFi.Ods.CodeGen
```
Comment (bptillman) [inline .github/workflows/Packages- EdFi.Ods.CodeGen.yml:1]:
```suggestion
name: Packages - EdFi.Ods.CodeGen
```
Please update this and the filename as follows to match the pattern we are setting up for these builds. It's just missing a space before the dash.

File: .github/workflows/Packages- EdFi.Ods.CodeGen.yml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        default: 'warning'
+
+env:
+  INFORMATIONAL_VERSION: "6.1"
+  BUILD_INCREMENTER: "160"
+  CONFIGURATION: "Release"
+  AZURE_ARTIFACT_URL: "https://pkgs.dev.azure.com/ed-fi-alliance/Ed-Fi-Alliance-OSS/_packaging/EdFi/nuget/v3/index.json"
+  AZURE_ARTIFACT_NUGET_KEY: ${{ secrets.AZURE_ARTIFACTS_PERSONAL_ACCESS_TOKEN }}
+  VSS_NUGET_EXTERNAL_FEED_ENDPOINTS : '{"endpointCredentials": [{"endpoint": "https://pkgs.dev.azure.com/ed-fi-alliance/Ed-Fi-Alliance-OSS/_packaging/EdFi/nuget/v3/index.json","password": "${{ secrets.AZURE_ARTIFACTS_PERSONAL_ACCESS_TOKEN }}"}]}'
+  CURRENT_BRANCH:  ${{ GITHUB.HEAD_REF }}
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply .github/workflows/Packages- EdFi.Ods.CodeGen.yml]: Fixed it . Thanks

File: .github/workflows/Packages- EdFi.Ods.CodeGen.yml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            $is_pull_request_branch = 'False'
+            $is_pull_request_branch = git ls-remote --heads origin ${{ env.CURRENT_BRANCH }} | Select-String -Pattern $patternName -SimpleMatch -Quiet
+            if ($is_pull_request_branch -eq $true) {
+              git fetch origin ${{ env.CURRENT_BRANCH }}
+              git checkout ${{ env.CURRENT_BRANCH }}
+            }                  
+    - name: build
+      working-directory: ./Ed-Fi-ODS/
+      run: |
+        .\build.ps1 build -Configuration ${{ env.CONFIGURATION }} -InformationalVersion ${{ env.INFORMATIONAL_VERSION}} -BuildCounter ${{ github.run_number }} -BuildIncrementer ${{env.BUILD_INCREMENTER}} -Solution "$env:GITHUB_WORKSPACE/Ed-Fi-ODS/Utilities/CodeGeneration/EdFi.Ods.CodeGen.sln"
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply .github/workflows/Packages- EdFi.Ods.CodeGen.yml:89]: yes  I added --version-suffix parameter

File: .github/workflows/Packages- EdFi.Ods.CodeGen.yml (hunk 1 above)
Comment (semalaiappan) [reply .github/workflows/Packages- EdFi.Ods.CodeGen.yml]: Fixed it . Thanks

**Review by bptillman — APPROVED** [review]
Approving this. The only failing build is TeamCity due to a conflict on the package published there already existing so nothing related to the functionality of this build.

**Pull request merged**